posts/2025-01-09-Rust-1.84.0.md

```diff
@@ -0,0 +1,110 @@
+---
+layout: post
+title: "Announcing Rust 1.84.0"
+author: The Rust Release Team
+release: true
+---
+
+The Rust team is happy to announce a new version of Rust, 1.84.0. Rust is a programming language empowering everyone to build reliable and efficient software.
+
+If you have a previous version of Rust installed via `rustup`, you can get 1.84.0 with:
+
+```console
+$ rustup update stable
+```
+
+If you don't have it already, you can [get `rustup`](https://www.rust-lang.org/install.html) from the appropriate page on our website, and check out the [detailed release notes for 1.84.0](https://doc.rust-lang.org/stable/releases.html#version-1840-2025-01-09).
+
+If you'd like to help us out by testing future releases, you might consider updating locally to use the beta channel (`rustup default beta`) or the nightly channel (`rustup default nightly`). Please [report](https://github.com/rust-lang/rust/issues/new/choose) any bugs you might come across!
+
+## What's in 1.84.0 stable
+
+### Cargo can use toolchain version for library version selection
+
+1.84.0 stabilizes the minimum supported Rust version (MSRV) aware resolver,
+which uses the declared [minimum supported Rust version](https://doc.rust-lang.org/cargo/reference/rust-version.html) from
+dependencies, if available, to improve package version selection. Rust version
+aware selection allows library authors to easily adopt newer Rust versions
+while providing library consumers a way to opt-in to using new package versions
+if they need compatibility with older toolchains.
+
+Library authors should start declaring their MSRV. Previously, bumping the
+version at each release to latest stable would force downstream consumers
+requiring an older Rust version to avoid running `cargo update` and/or require
+pinning the version of the library in use, but now those consumers will be able
+to automatically avoid pulling in new library versions if they want
+compatibility with an older toolchain. We expect this to provide more options
+for library authors for picking their preferred support strategy for Rust
+versions.
+
+The new resolver will be enabled by default with the 2024 edition (expected to
+stabilize in 1.85), but can be enabled as of 1.84 by setting
+`resolver.incompatible-rust-versions = "fallback"`.
```

Comment:
To me, setting it in manifest is a bit better than in config file. Not every project checks in `.cargo/config.toml`.

```suggestion
[`package.resolver = "3"`](https://doc.rust-lang.org/cargo/reference/resolver.html#resolver-versions) in the Cargo.toml manifest file, or
[`resolver.incompatible-rust-versions = "fallback"`](https://doc.rust-lang.org/cargo/reference/config.html#resolverincompatible-rust-versions) in the Cargo configuration file.
```